clang/test/Driver/mcmodel.c

```diff
@@ -13,6 +13,11 @@
 // RUN: not %clang -### -c --target=aarch64 -mcmodel=medium %s 2>&1 | FileCheck --check-prefix=ERR-MEDIUM %s
 // RUN: not %clang -### -c --target=aarch64 -mcmodel=kernel %s 2>&1 | FileCheck --check-prefix=ERR-KERNEL %s
 // RUN: not %clang --target=aarch64_32-linux -### -S -mcmodel=small %s 2>&1 | FileCheck --check-prefix=ERR-AARCH64_32 %s
+// RUN: %clang --target=amdgcn-amd-amdhsa -nogpulib -### -c -mcmodel=tiny %s 2>&1 | FileCheck --check-prefix=AMDGPU-MCMODEL-TINY %s
```

Comment:
I'm not sure exactly what's being tested here, this looks like it's targeting C as a target? The more relevant case would be with amdgcn as the offload target. For direct compilation it would make more sense to emit the error

Reply:
The change is intended to handle all possible code models for AMDGPU target. Basically, we want to avoid erroring out for device side code models. Does it make sense to have negative case for amdgpu ?

Reply:
There's 2 cases. The case where we're directly codegening to the target, in which case it behaves like a regular CPU target. That's what you're testing here, but it's somewhat unusual. We could error here as before.

The more realistic case is treating it as an offload target from OpenMP or HIP, so I think it's more important to test that case. We definitely don't want an error here.

Reply:
I think, recent updates to PR (3rd commit: 9dc7689983b74dd381df556bc9e61165e08d85b4) addresses both the cases you are referring to.